framework/source/class/qx/lang/Type.js

```diff
@@ -101,7 +101,9 @@ qx.Bootstrap.define("qx.lang.Type",
       // JavaScript null and a null returned from calling DOM.
       // e.q. by document.getElementById("ReturnedNull").
       return (
-        value !== null && (
+        value !== null &&
+        // Check if value is NaN
+        value === value && (
```

Comment:
The idea was that since this is a general purpose function the value could be even an empty string for which isNaN("") returns false. This would serve the purpose anyway and if something went wrong there was still the last part of the && clause to pick it up. Still it felt less predictable than equality check.